src/Signup_Command.php

```diff
@@ -0,0 +1,256 @@
+				// Support features like --active=0.
+				foreach ( array_keys( $item ) as $field ) {
+					if ( isset( $assoc_args[ $field ] ) && $assoc_args[ $field ] !== $item[ $field ] ) {
+						continue 2;
+					}
+				}
+
+				$signups[] = $item;
+			}
+		}
+
+		$format = Utils\get_flag_value( $assoc_args, 'format', 'table' );
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		if ( 'ids' === $format ) {
+			WP_CLI::line( implode( ' ', wp_list_pluck( $signups, 'signup_id' ) ) );
+		} else {
+			$formatter->display_items( $signups );
+		}
+	}
+
+	/**
+	 * Gets details about the signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : Signup ID, user login, user email or activation key.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole signup, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Get a specific subset of the signup's fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get signup.
+	 *     $ wp signup get 1
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | signup_id | user_login | user_email          | registered          | active | activation_key   |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | 1         | bobuser    | bobuser@example.com | 2024-03-13 05:46:53 | 1      | 663b5af63dd930fd |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *
+	 * @package wp-cli
+	 */
+	public function get( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$formatter->display_items( array( $signup ) );
+	}
+
+	/**
+	 * Activates a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : Signup ID, user login, user email or activation key.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Activate signup.
+	 *     $ wp signup activate 2
+	 *     Success: Signup activated. Password: bZFSGsfzb9xs
+	 *
+	 * @package wp-cli
+	 */
+	public function activate( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		if ( $signup ) {
+			$result = wpmu_activate_signup( $signup->activation_key );
+		}
+
+		if ( is_wp_error( $result ) ) {
+			WP_CLI::error( 'Signup could not be activated. Reason: ' . $result->get_error_message() );
+		} else {
+			WP_CLI::success( "Signup activated. Password: {$result['password']}" );
+		}
+	}
+
+	/**
+	 * Deletes a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : Signup ID, user login, user email or activation key.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Delete signup.
+	 *     $ wp signup delete 3
+	 *     Success: Signup deleted.
+	 *
+	 * @package wp-cli
+	 */
+	public function delete( $args, $assoc_args ) {
```

Comment:
Does it make sense to use the `_delete()` method with a `delete_callback()` ([example](https://github.com/wp-cli/entity-command/blob/22d726dd3ce09a7a70ca1ac2c935e2f0450894a4/src/Post_Command.php#L493-L523))?

Reply:
Do we really need this `_delete()` approach? I checked this `_delete` function and it is used to enable/disable `wp_defer_term_counting()` which signup has no relation.

Reply:
If I understand correctly, the `_delete()` callback can contain anything. But I understand if you think this is unnecessary, given the few lines of code. It would have aligned the code structure to similar commands, like the mentioned `wp post delete`. 🙂

Reply:
Thanks for doing this! 🙏🏽